R/derive_locf_records.R

```diff
@@ -93,25 +104,39 @@
 #'   data = advs,
 #'   dataset_expected_obs = advs_expected_obsv,
 #'   by_vars = exprs(STUDYID, USUBJID, PARAMCD),
-#'   order = exprs(AVISITN, AVISIT)
+#'   order = exprs(AVISITN, AVISIT),
+#'   keep_vars = exprs(PARAMN)
 #' )
 #'
 derive_locf_records <- function(dataset,
                                 dataset_expected_obs,
                                 by_vars,
-                                order) {
+                                analysis_var = AVAL,
+                                order,
+                                keep_vars = NULL) {
   #### Input Checking ####
+  analysis_var <- assert_symbol(enexpr(analysis_var))
 
   # Check if input parameters is a valid list of variables
   assert_vars(by_vars, optional = TRUE)
+  assert_vars(keep_vars, optional = TRUE)
   assert_order_vars(order)
 
   # Check by_vars and order variables in input datasets
   assert_data_frame(dataset_expected_obs)
   assert_data_frame(
     dataset,
-    required_vars = expr_c(by_vars, extract_vars(order), chr2vars(colnames(dataset_expected_obs)))
+    required_vars = expr_c(
+      by_vars, analysis_var, order,
```

Comment:
Please use `extract_vars(order)` instead of `order`.  Otherwise the code fails if order is something like `desc(ASEQ)`.